cli/command/image/build_session_test.go

```diff
@@ -0,0 +1,37 @@
+package image
+
+import (
+	"bytes"
+	"path/filepath"
+	"testing"
+
+	"github.com/docker/docker/pkg/streamformatter"
+	"github.com/gotestyourself/gotestyourself/fs"
+	"github.com/moby/buildkit/session"
+	"github.com/stretchr/testify/require"
+)
+
+func TestAddDirToSession(t *testing.T) {
+	dest := fs.NewDir(t, "test-build-session",
+		fs.WithFile("Dockerfile", `
+			FROM alpine:3.6
+			COPY foo /
+		`),
+		fs.WithFile("foo", "some content", fs.AsUser(65534, 65534)),
+	)
+	defer dest.Remove()
+
+	contextDir := dest.Path()
+	sharedKey, err := getBuildSharedKey(contextDir)
+	require.NoError(t, err)
+
+	var s *session.Session
+	s, err = session.NewSession(filepath.Base(contextDir), sharedKey)
```

Comment:
@dnephin @tonistiigi ok I guess the best is to mock/fake the session here instead of using the true one. @tonistiigi does it worth that builkit provides a session faker at some point? I can put it in the cli ATM.

Reply:
Having the CLI depend on a `type Session interface{...}` instead of the struct would  be great, then it should be easy for us to fake.

A fake as part of the CLI testing code sounds fine to me.